Tensile/Source/lib/include/Tensile/MasterSolutionLibrary.hpp

```diff
@@ -84,7 +84,17 @@ namespace Tensile
                                                              double*          fitness
                                                              = nullptr) const override
         {
-            return library->findBestSolution(problem, hardware, fitness);
+            const int solution_index = Debug::Instance().getSolutionIndex();
+
+            if(solution_index >= 0)
+            {
+                std::cout << "Tensile will use solution index: " << solution_index <<std::endl;
+                std::cout << "Warning: Tensile will only work for a particular transpose and data type." << std::endl;
+                std::cout << "Set TENSILE_SOLUTION_INDEX to a negative number to restore the default behavior." << std::endl;
+                return solutions.at(solution_index);
```

Comment:
You still need to check the predicates. `(*solution->problemPredicate)(problem)` and `(*solution->hardwarePredicate)(hardware)` must both return `true` or we should return `nullptr` from here.